File: app/java/app/src/main/java/com/example/accounting/domain/event/AccountDeletedEvent.java
package com.example.accounting.domain.event;

import lombok.Getter;

import java.util.Map;

/**
 * 勘定科目削除イベント
 */
@Getter
public class AccountDeletedEvent extends DomainEvent {
    private final String accountCode;
    private final Map<String, Object> oldValues;
    private final String reason;
    private final String userId;
    private final String userName;
    private final String ipAddress;

    public AccountDeletedEvent(
            String accountCode,
            Map<String, Object> oldValues,
            String reason,
            String userId,
            String userName,
            String ipAddress) {
        super();
        this.accountCode = accountCode;
        this.oldValues = oldValues;
        this.reason = reason;
        this.userId = userId;
        this.userName = userName;
        this.ipAddress = ipAddress;
    }

    @Override
    public String getEventType() {
        return "AccountDeleted";
    }
}

File: app/csharp/AccountingSystem.Domain/Events/AccountDeletedEvent.cs
namespace AccountingSystem.Domain.Events;

/// <summary>
/// 勘定科目削除イベント
/// </summary>
public record AccountDeletedEvent : DomainEvent
{
    public required string AccountCode { get; init; }
    public required Dictionary<string, object> DeletedData { get; init; }
    public required string UserId { get; init; }
    public required string UserName { get; init; }
    public string? Reason { get; init; }
    public string? IpAddress { get; init; }

    public override string GetEventType() => "AccountDeleted";
}
